src/main/java/io/codelex/TestAdvanced/Exercise1/DebitCard.java

```diff
@@ -0,0 +1,29 @@
+package io.codelex.TestAdvanced.Exercise1;
+
+public class DebitCard extends Card {
+
+    private static final int MAX_BALANCE = 10000;
+
+    public DebitCard(int number, String fullName, double ccv, double balance) {
+        super(number, fullName, ccv, balance);
+    }
+
+    @Override
+    public void addMoney(double amount) {
+        setBalance(getBalance() + amount);
+        if (getBalance() > MAX_BALANCE) {
+            System.out.println("Warning: Too much money");
+        }
+    }
+
+    @Override
+    public void takeMoney(double amount) throws NotEnoughFundsException {
+
+        try {
+            setBalance(getBalance() - amount);
+        } catch (ArithmeticException e) {
```

Comment:
Te faktiski tas pats - man īsti nav skaidrs kāpēc te ir ArithmeticException, jo kods, kas ir try blokā tādu nevar mest. Te idejiski jābūt:
```
if (balance < amount) {
   mest kļūdu
}
setBalance(getBalance() - amount);
```
Tad kad izdarīsi izmaiņas, tad varēsi redzēt, ka abām kartēm ir kopīga daļa no takeMoney. To var pārnest uz Card klasi.

Reply:
Hey, salaboju.